backend/src/main/java/fk/prof/backend/model/assignment/AggregationWindowPlanner.java

```diff
@@ -0,0 +1,203 @@
+package fk.prof.backend.model.assignment;
+
+import com.google.common.base.Preconditions;
+import fk.prof.aggregation.model.FinalizedAggregationWindow;
+import fk.prof.backend.aggregator.AggregationWindow;
+import fk.prof.backend.model.aggregation.AggregationWindowLookupStore;
+import fk.prof.backend.proto.BackendDTO;
+import fk.prof.backend.util.BitOperationUtil;
+import fk.prof.backend.util.proto.RecorderProtoUtil;
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import recording.Recorder;
+
+import java.time.Clock;
+import java.time.LocalDateTime;
+import java.util.Random;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class AggregationWindowPlanner {
+  private static int workIdCounter = 1;
+  private static final int MILLIS_IN_SEC = 1000;
+  private static final Logger logger = LoggerFactory.getLogger(AggregationWindowPlanner.class);
+  //Should be a different value if backend were to die and come back up, otherwise duplicate work ids will be generated by the backend since workId counter starts from 0 on every process start
+  public static final int BACKEND_IDENTIFIER;
+
+  static {
+    Random random = new Random();
+    //Random does not guarantee unique values every single time but should be an acceptable(and minimal) risk since we have 32 bits to play with here
+    BACKEND_IDENTIFIER = random.nextInt();
```

Comment:
Agreed, but how will ansible generate this. UUID won't cut it, since work id is 64 bits long, we need a 16 bit or max 32 bit unique identifier